tsconfig.json

```diff
@@ -1,17 +1,22 @@
 {
   "compilerOptions": {
-    "moduleResolution": "node",
     "allowJs": true,
-    "module": "esnext",
-    "target": "es2017",
-    "strict": true,
-    "noEmitOnError": false,
-    "sourceMap": false,
-    "declaration": false,
     "allowSyntheticDefaultImports": true,
+    "declaration": false,
+    "exactOptionalPropertyTypes": true,
     "experimentalDecorators": true,
     "emitDecoratorMetadata": true,
-    "outDir": "dist/src"
+    "module": "esnext",
+    "moduleResolution": "node",
+    "noEmitOnError": false,
+    "noImplicitOverride": true,
+    "noPropertyAccessFromIndexSignature": true,
+    "noUncheckedIndexedAccess": true,
+    "outDir": "dist/src",
+    "sourceMap": false,
+    "strict": true,
+    "target": "es2017",
```

Comment:
should we think about bumping this up?

Reply:
Since the project still supports Node v10 (and we really should remove that support at some point), I'm not sure how much we can bump this ATM. Node v10 was released on 2018-04-24, which is right is _probably_ right for now? I haven't tested it myself too thoroughly though TBH

Reply:
ahh node 10...well in that case it makes sense to keep this as-is for now anyway